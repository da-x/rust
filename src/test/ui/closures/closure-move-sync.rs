use std::thread;
use std::sync::mpsc::channel;

fn bar() {
    let (send, recv) = channel();
    let t = thread::spawn(|| {
        recv.recv().unwrap();
        //~^^ ERROR `Receiver<()>` cannot be shared between threads safely
    });

    send.send(());

    t.join().unwrap();
}

fn foo() {
    let (tx, _rx) = channel();
    thread::spawn(|| tx.send(()).unwrap());
    //~^ ERROR `Sender<()>` cannot be shared between threads safely
}

fn main() {}
