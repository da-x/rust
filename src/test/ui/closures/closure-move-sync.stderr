error[E0277]: `Receiver<()>` cannot be shared between threads safely
  --> $DIR/closure-move-sync.rs:6:13
   |
LL |     let t = thread::spawn(|| {
   |             ^^^^^^^^^^^^^ `Receiver<()>` cannot be shared between threads safely
   | 
  ::: $SRC_DIR/std/src/thread/mod.rs:LL:COL
   |
LL |     F: Send + 'static,
   |        ---- required by this bound in `spawn`
   |
   = help: the trait `Sync` is not implemented for `Receiver<()>`
   = note: required because of the requirements on the impl of `Send` for `&Receiver<()>`
   = note: required because it appears within the type `[closure@$DIR/closure-move-sync.rs:6:27: 9:6 recv:&Receiver<()>]`

error[E0277]: `Sender<()>` cannot be shared between threads safely
  --> $DIR/closure-move-sync.rs:18:5
   |
LL |     thread::spawn(|| tx.send(()).unwrap());
   |     ^^^^^^^^^^^^^ `Sender<()>` cannot be shared between threads safely
   | 
  ::: $SRC_DIR/std/src/thread/mod.rs:LL:COL
   |
LL |     F: Send + 'static,
   |        ---- required by this bound in `spawn`
   |
   = help: the trait `Sync` is not implemented for `Sender<()>`
   = note: required because of the requirements on the impl of `Send` for `&Sender<()>`
   = note: required because it appears within the type `[closure@$DIR/closure-move-sync.rs:18:19: 18:42 tx:&Sender<()>]`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
